receiver/splunkhecreceiver/receiver.go

```diff
@@ -295,18 +379,40 @@ func (r *splunkReceiver) handleRawReq(resp http.ResponseWriter, req *http.Reques
 		r.failRequest(ctx, resp, http.StatusInternalServerError, errInternalServerError, slLen, err)
 		return
 	}
+
 	consumerErr := r.logsConsumer.ConsumeLogs(ctx, ld)
 
 	_ = bodyReader.Close()
 
 	if consumerErr != nil {
 		r.failRequest(ctx, resp, http.StatusInternalServerError, errInternalServerError, slLen, consumerErr)
 	} else {
-		r.writeSuccessResponse(ctx, resp, ld.LogRecordCount())
+		if r.ackExt != nil {
+			r.writeSuccessResponseWithAck(ctx, resp, ld.LogRecordCount(), partitionID)
+		} else {
+			r.writeSuccessResponseWithoutAck(ctx, resp, ld.LogRecordCount())
+		}
 		r.obsrecv.EndLogsOp(ctx, metadata.Type.String(), slLen, nil)
 	}
 }
 
+func (r *splunkReceiver) validateChannelHeader(ctx context.Context, resp http.ResponseWriter, req *http.Request) string {
```

Comment:
nit: function is technically doing `validateAndExtractChannelHeader`

Reply:
I also find this method a bit weird. it is handling error cases but not returning errors. Do you think this will be better?

```
func (r *splunkReceiver) validateAndExtractChannelHeader(ctx context.Context, resp http.ResponseWriter, req *http.Request) (string, error) {
        // check channel header exists
	partitionID, ok := req.Header.Get(splunk.HTTPSplunkChannelHeader)
	if !ok {
	    return partitionID, ok // no channel ID
	}
	
	if len(partitionID) == 0 {
		return "", errors.New(responseErrDataChannelMissing)
	}
	// check validity of channel
	_, err := uuid.Parse(partitionID)
	if err != nil {
		return "", errors.New(responseErrInvalidDataChannel)
	}

	return partitionID, nil
```

On the caller side:
```
var partitionID string
	if r.ackExt != nil {
                var err error
		if partitionID, error = r. validateAndExtractChannelHeader(ctx, resp, req); err != nil {
		    r.failRequest(ctx, resp, http.StatusBadRequest, err.Error(), 0, err)
		}
	}
```